indica/readers/abstractreader.py

```diff
@@ -84,985 +44,187 @@ def __init__(
+        self.quantities_path = self.machine_conf.QUANTITIES_PATH
+        self.default_error = default_error
+        self.verbose = verbose
+        self.return_dataarrays = return_dataarrays
+        self.kwargs = kwargs
 
     def get(
         self,
         uid: str,
         instrument: str,
         revision: RevisionLike = 0,
-        quantities: Set[str] = set(),
-        **kwargs,
+        dl: float = 0.005,
+        passes: int = 1,
+        include_error: bool = True,
     ) -> Dict[str, DataArray]:
-        """Reads data for the requested instrument. In general this will be
-        the method you want to use when reading.
-
-        Parameters
-        ----------
-        uid
-            User ID (i.e., which user created this data)
-        instrument
-            The instrument which measured this data (DDA at JET)
-        revision
-            An object (of implementation-dependent type) specifying what
-            version of data to get. Default is the most recent.
-        quantities
-            Which physical quantitie(s) to read from the database. Defaults to
-            all available quantities for that instrument.
-
-        Returns
-        -------
-        :
-            A dictionary containing the requested physical quantities.
-        """
-        if instrument not in self.INSTRUMENT_METHODS:
+        """General method that reads data for a requested instrument."""
+        if instrument not in self.instrument_methods.keys():
             raise ValueError(
                 "{} does not support reading for instrument {}".format(
                     self.__class__.__name__, instrument
                 )
             )
-        method = getattr(self, self.INSTRUMENT_METHODS[instrument])
-        if not quantities:
-            quantities = set(self.available_quantities(instrument))
-        return method(uid, instrument, revision, quantities, **kwargs)
-
-    def get_thomson_scattering(
-        self,
-        uid: str,
-        instrument: str,
-        revision: RevisionLike,
-        quantities: Set[str],
-        **kwargs,
-    ) -> Dict[str, DataArray]:
-        """
-        Reads data based on Thomson Scattering.
-        """
-        database_results = self._get_thomson_scattering(
-            uid, instrument, revision, quantities
-        )
-
-        transform = TransectCoordinates(
-            database_results["x"],
-            database_results["y"],
-            database_results["z"],
-            f"{instrument}",
-            machine_dimensions=database_results["machine_dims"],
-        )
-        database_results["channel"] = np.arange(database_results["length"])
+        # Read data from database
+        data = self._read_database(uid, instrument, revision)
+
+        # Re-arrange data (machine-specific)
+        method = self.instrument_methods[instrument]
+        data = getattr(self, f"_{method}")(data)
+
+        if self.return_dataarrays:
```

Comment:
I don't think the base class should need to know details about what transforms are used. Ideally the base class would not need to know anything about the data since it is so general.

Reply:
done